src/librustc/Cargo.toml

```diff
@@ -33,6 +33,7 @@ parking_lot = "0.5.5"
 byteorder = { version = "1.1", features = ["i128"]}
 chalk-engine = { version = "0.7.0", default-features=false }
 rustc_fs_util = { path = "../librustc_fs_util" }
+smallvec = "0.6.4"
```

Comment:
We may want to use the `union` feature here. (`smallvec = { version = "0.6.4", features = ["union"] }`).